crates/volta-core/src/run/pnpm.rs

```diff
@@ -0,0 +1,73 @@
+use std::env;
+use std::ffi::OsString;
+
+use super::executor::{Executor, ToolCommand, ToolKind};
+use super::parser::CommandArg;
+use super::{debug_active_image, debug_no_platform, RECURSION_ENV_VAR};
+use crate::error::{ErrorKind, Fallible};
+use crate::platform::{Platform, Source, System};
+use crate::session::{ActivityKind, Session};
+
+pub(super) fn command(args: &[OsString], session: &mut Session) -> Fallible<Executor> {
+    session.add_event_start(ActivityKind::Pnpm);
+    // Don't re-evaluate the context or global install interception if this is a recursive call
+    let platform = match env::var_os(RECURSION_ENV_VAR) {
+        Some(_) => None,
+        None => {
+            // FIXME: Figure out how to intercept pnpm global commands properly.
+            // This guard prevents all global commands from running, it should
+            // be removed when we fully implement global command interception.
+            let is_global = args.iter().any(|f| f == "--global" || f == "-g");
+            if is_global {
+                return Err(ErrorKind::Unimplemented {
+                    feature: "pnpm global commands".into(),
+                }
+                .into());
+            }
+
+            if let CommandArg::Global(cmd) = CommandArg::for_pnpm(args) {
+                // For globals, only intercept if the default platform exists
+                if let Some(default_platform) = session.default_platform()? {
+                    return cmd.executor(default_platform);
+                }
+            }
```

Comment:
Since we're guarding against global commands above anyway, could we remove this block for now? If necessary, we can add a `#[allow(dead_code)]` directive to the `CommandArg::for_pnpm` method, to prevent warnings about it being unused.